ts/Data/DataTable.ts

```diff
@@ -880,9 +881,9 @@ class DataTable extends DataTableCore implements DataEvent.Emitter {
             if (Array.isArray(column)) {
                 // Normal array
                 rowIndex = column.indexOf(cellValue, rowIndexOffset);
-            } else if (defined(cellValue) && Number.isFinite(cellValue)) {
+            } else if (isNumber(cellValue)) {
                 // Typed array
-                rowIndex = column.indexOf(+cellValue, rowIndexOffset);
+                rowIndex = column.indexOf(cellValue, rowIndexOffset);
             }
```

Comment:
The code should theoretically work with this simplification:

```ts
if (Array.isArray(column) || isNumber(cellValue)) {
    rowIndex = column.indexOf(cellValue, rowIndexOffset);
}
```

but unfortunately, TypeScript doesn’t allow `cellValue` in this case:

![image](https://github.com/user-attachments/assets/87985590-9e4d-4ef9-8a08-f351a295853f)